sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1479,7 +1479,13 @@ class SessionCatalog(
     requireDbExists(db)
     val newFuncDefinition = funcDefinition.copy(identifier = qualifiedIdent)
     if (!functionExists(qualifiedIdent)) {
-      externalCatalog.createFunction(db, newFuncDefinition)
+      try {
+        externalCatalog.createFunction(db, newFuncDefinition)
+      } catch {
+        case e: FunctionAlreadyExistsException if ignoreIfExists =>
+          logWarning(s"Function ${qualifiedIdent.funcName} already exists in database $db. " +
```

Comment:
for the happy path we do nothing, shall we skip the logging here as this is not a bad case? user specified `IF NOT EXIST`